ampd/src/grpc/reqs.rs

```diff
@@ -24,12 +26,40 @@ pub fn validate_broadcast(req: Request<BroadcastRequest>) -> Result<Any, Error>
         .ok_or(report!(Error::EmptyBroadcastMsg))
 }
 
+pub fn validate_smart_contract_state(
+    req: Request<SmartContractStateRequest>,
+) -> Result<(TMAddress, Vec<u8>), Error> {
+    let SmartContractStateRequest { contract, query } = req.into_inner();
+
+    ensure!(!query.is_empty(), Error::InvalidQuery);
+    let query: serde_json::Value =
+        serde_json::from_slice(&query).change_context(Error::InvalidQuery)?;
+
+    Ok((
+        validate_address(&contract)?,
+        serde_json::to_vec(&query).expect("must be valid JSON"),
+    ))
+}
+
+fn validate_address(address: &str) -> Result<TMAddress, Error> {
+    let address = TMAddress::from_str(address)
+        .change_context(Error::InvalidContractAddress(address.to_string()))?;
+    ensure!(
+        address.prefix() == PREFIX,
+        Error::InvalidContractAddress(address.to_string())
+    );
+
+    Ok(address)
+}
+
 #[derive(Error, Debug)]
 pub enum Error {
     #[error("empty filter")]
     EmptyFilter,
-    #[error("invalid contract address in filter")]
+    #[error("invalid contract address")]
     InvalidContractAddress(String),
```

Comment:
what happens to this address string?

Reply:
It's used here https://github.com/axelarnetwork/axelar-amplifier/blob/704e1f04de2a76cb55bdc6e11d0ef6cae489de67/ampd/src/grpc/error.rs#L66